src/query/functions-v2/src/scalars/variant.rs

```diff
@@ -40,4 +75,731 @@ pub fn register(registry: &mut FunctionRegistry) {
             Ok(())
         }),
     );
+
+    registry.register_passthrough_nullable_1_arg::<BooleanType, VariantType, _, _>(
+        "parse_json",
```

Comment:
I was confused before but yet has no chance to make a question that 

When clickhouse adopts `CAST(... AS VARIANT)`, `TRY_CAST(... AS VARIANT)` style conversions, and snowflake adopts `parse_json(...)` and `try_parse_json(...)` style conversions, which style should we pick?

AFAICS, the `CAST` style works better with strong typed `Array` because it allows to `CAST(Array(Type) AS Array(VARIANT))`, while the `parse_json(...)` style allows users to give additional parameters (`parse_json` in snowflake doesn't  have extra parameters so I'll take `to_timestamp` as an example: [`to_timestamp(<string>, <format>)`](https://docs.snowflake.com/en/sql-reference/functions/to_timestamp.html#:~:text=timestampFunction%20(%20%3Cstring_expr%3E%20%5B%20%2C%20%3Cformat%3E%20%5D%20)) allows to specify the string format while it will be hard for `CAST(<string> AS TIMESTAMP)`)

TBO, it is sort of a mass in databend. Maybe we could discuss it before pushing forward.

cc @sundy-li

Reply:
> Do they duplicate with `CAST(... AS VARIANT)`, which will call t[his conversion function](https://github.com/datafuselabs/databend/blob/df03f06e55e6ea2c7d05fd2d745e9e6b692726c7/src/query/expression/src/types/variant.rs#L160)?

yes, we can remove `Boolean` and `Number`.

Reply:
I think we can keep them both, `CAST(... AS VARIANT)` is used to convert basic data types to `Variant`, and `parse_json(...)` is used to convert a `JSON` string to `Variant`.

Reply:
I think `parse_json` can be treated as another function that converts a `JSON` string to `Variant`. It's different from standard cast rules.